lib/nmatrix/nmatrix.rb

```diff
@@ -90,6 +90,62 @@ def size(shape)
       shape = [shape,shape] unless shape.is_a?(Array)
       (0...shape.size).inject(1) { |x,i| x * shape[i] }
     end
+
+    # Make N-D coordinate arrays for vectorized evaluations of
+    # N-D scalar/vector fields over N-D grids, given N one-dimensional
+    # coordinate arrays arrs. N > 1.
+    #
+    # call-seq:
+    #     meshgrid(arrs) -> Array of NMatrix
+    #     meshgrid(arrs, options) -> Array of NMatrix
+    #
+    # * *Arguments* :
+    #   - +arrs+ -> Array of N one-dimensional coordinate arrays (Array or NMatrix)
+    #   - +options+ -> Hash with options (:sparse and :ij both Boolean, false by default)
+    # * *Returns* :
+    #   - Array of N N-D NMatrixes
+    # * *Examples* :
+    #     x, y = NMatrix::meshgrid([[1, 2, 3], [4, 5]])
+    #     x.to_a #<= [[1, 2, 3], [1, 2, 3]]
+    #     y.to_a #<= [[4, 4, 4], [5, 5, 5]]
+    #
+    # * *Using* *options* :
+    #
+    #     x, y = NMatrix::meshgrid([[1, 2, 3], [4, 5]], sparse: true)
+    #     x.to_a #<= [[1, 2, 3]]
+    #     y.to_a #<= [[4], [5]]
+    #
+    #     x, y = NMatrix::meshgrid([[1, 2, 3], [4, 5]], ij: true)
+    #     x.to_a #<= [[1, 1], [2, 2], [3, 3]]
+    #     y.to_a #<= [[4, 5], [4, 5], [4, 5]]
+    def meshgrid(arrs, options = {})
+      raise(ArgumentError, 'Expected at least 2 arrays.') if arrs.size < 2
+      mats = arrs.map do |arr|
```

Comment:
I think you should raise an error if `:ij` and `:xy` are both specified as `true`.

Or instead of that, you can change the option called `:indexing` and let users specify the type of indexing with an `indexding: :ij` or `indexing: :xy` option, instead `:ij => true` or `:xy => true`.